packages/@lwc/engine-core/src/framework/base-bridge-element.ts

```diff
@@ -121,9 +122,26 @@ export function HTMLBridgeElementFactory(
     const { attributeChangedCallback: superAttributeChangedCallback } = SuperClass.prototype as any;
     const { observedAttributes: superObservedAttributes = [] } = SuperClass as any;
     const descriptors: PropertyDescriptorMap = create(null);
+
+    // present a hint message so that developers are aware that they have not decorated property with @api
+    if (process.env.NODE_ENV !== 'production') {
+        for (let i = 0, len = privateProperties.length; i < len; i += 1) {
+            const propName = privateProperties[i];
+            attributeToPropMap[htmlPropertyToAttribute(propName)] = propName;
+            descriptors[propName] = {
+                get() {
+                    logWarn(`Property ${propName} is not decorated with @api`);
```

Comment:
I'd love for this to have a bit more detail. E.g.:

```suggestion
                    logWarn(`The property "${propName}" is not publicly accessible. Add the @api annotation to the property declaration or getter/setter in the component to make it accessible.`);
```